src/lavinmq/amqp/queue/queue.cr

```diff
@@ -190,13 +190,39 @@ module LavinMQ::AMQP
         @paused.set(true)
       end
       @msg_store = init_msg_store(@data_dir)
+      @empty = @msg_store.empty
+      start
+    end
+
+    private def start
       if @msg_store.closed
```

Comment:
Why do we check if the msg_store is closed on start? From the code in MessageStore it looks like it's only set to closed = true on `close` and `delete`  so can this really happen here?

Reply:
yes, `MessageStore#close` can be called if there are errors in `load_segments_from_disk` or `produce_metadata` during start, and if that happens there's no point in trying to start the queue